crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs

```diff
@@ -131,6 +132,51 @@ impl DependencyScanner<'_> {
     }
   }
 
+  // new URL("./foo.png", import.meta.url);
+  fn add_new_url(&mut self, new_expr: &NewExpr, ast_path: &AstNodePath<AstParentNodeRef<'_>>) {
+    if let Expr::Ident(Ident {
+      sym: js_word!("URL"),
+      ..
+    }) = &*new_expr.callee
+    {
+      if let Some(args) = &new_expr.args {
+        if let (Some(first), Some(second)) = (args.first(), args.get(1)) {
+          if let (
+            ExprOrSpread {
+              spread: None,
+              expr: box Expr::Lit(Lit::Str(path)),
+            },
+            // import.meta.url
+            ExprOrSpread {
+              spread: None,
+              expr:
+                box Expr::Member(MemberExpr {
+                  obj:
+                    box Expr::MetaProp(MetaPropExpr {
+                      kind: MetaPropKind::ImportMeta,
+                      ..
+                    }),
+                  prop:
+                    MemberProp::Ident(Ident {
+                      sym: js_word!("url"),
+                      ..
+                    }),
+                  ..
+                }),
+            },
+          ) = (first, second)
+          {
+            self.add_dependency(box NewUrlDependency::new(
```

Comment:
Can you port the relativeUrl part too? See: https://github.com/webpack/webpack/pull/12695

Reply:
This can be done in another PR